### PR TITLE
feat: platform independent server

### PR DESCRIPTION
**Pull request by manucorporat**

### CODE REVIEW

File: starters/apps/todo/assets/asset-1a6812a1 [hunk 1]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (mhevery) [inline starters/apps/todo/assets/asset-1a6812a1:1]: I don't believe this file belongs here.

File: starters/apps/todo/workers-site/assets/asset-1a6812a1 [hunk 2]
```diff
@@ -0,0 +1,20 @@
+{
```
Comment (mhevery) [inline starters/apps/todo/workers-site/assets/asset-1a6812a1:1]: revert?

File: starters/apps/todo/workers-site/index.js [hunk 3]
```diff
@@ -0,0 +1,12297 @@
+'use strict';
```
Comment (mhevery) [inline starters/apps/todo/workers-site/index.js]: This file should net be here.

**Comment by manucorporat** [comment]
Ready for review!

File: scripts/submodule-server.ts [hunk 4]
```diff
@@ -59,12 +61,11 @@ export async function submoduleServer(config: BuildConfig) {
     plugins: [
       importPath(/^@builder\.io\/qwik$/, '../core.cjs'),
       importPath(/^@builder\.io\/qwik\/optimizer$/, '../optimizer.cjs'),
-      dominoPlugin,
```
Comment (adamdbradley) [inline scripts/submodule-server.ts:62]: Domino should stay in the build right?

File: src/core/api.md [hunk 5]
```diff
@@ -4,6 +4,11 @@
 
 ```ts
 
+import type { CorePlatform as CorePlatform_2 } from '@builder.io/qwik';
```
Comment (adamdbradley) [inline src/core/api.md:7]: Not a blocker, but why is this happening? `CorePlatform as CorePlatform_2`

File: src/optimizer/core/src/transform.rs [hunk 6]
```diff
@@ -373,6 +373,18 @@ fn create_inline_qhook(url: JsWord, symbol: &str) -> ast::CallExpr {
                     kind: ast::StrKind::Synthesized,
                 }))),
             },
+            // ast::ExprOrSpread {
```
Comment (adamdbradley) [inline src/optimizer/core/src/transform.rs:376]: Just remove?

File: src/optimizer/src/rollup/index.ts [hunk 7]
```diff
@@ -144,19 +134,14 @@ export function qwikRollup(opts: QwikPluginOptions = {}): any {
...
-              type: 'asset',
-            });
+        if (typeof opts.symbolsOutput === 'string') {
+          this.emitFile({
+            fileName: opts.symbolsOutput,
+            source: JSON.stringify(outputEntryMap, null, 2),
+            type: 'asset',
           });
-          writeSymbols();
+        } else {
```
Comment (adamdbradley) [inline src/optimizer/src/rollup/index.ts:143]: Maybe double check the type is a `function`

File: src/server/api.md [hunk 8]
```diff
@@ -4,9 +4,8 @@
 
 ```ts
 
-/// <reference types="node" />
```
Comment (adamdbradley) [inline src/server/api.md:7]: Nice! Something is right if `/// <reference types="node" />` is removed. Love it.

File: src/server/prefetch.ts [hunk 9]
```diff
@@ -45,6 +40,7 @@ export async function getImports(
   filePath: string,
   readFileFn: (path: string) => Promise<string> = readFile
 ): Promise<string[]> {
+  const { join } = await import('path');
```
Comment (adamdbradley) [inline src/server/prefetch.ts:43]: Don't we have our own path utils now?

File: src/server/serialize.ts [hunk 10]
```diff
@@ -26,5 +31,26 @@ export function serializeDocument(doc: Document, opts?: SerializeDocumentOptions
...
+function createQrlMapper(qEntryMap: OutputEntryMap) {
+  const symbolManifest = new Map<string, string>();
+
+  const qrlMapper: QrlMapper = (path, symbolName) => {
+    path = symbolManifest.get(symbolName) || path;
+    path = path.slice(0, path.lastIndexOf('.'));
+    return `./${path}#${symbolName}`;
+  };
+
+  for (const symbolName in qEntryMap.mapping) {
```
Comment (adamdbradley) [inline src/server/serialize.ts:47]: Wonder if we should throw an error here if the object is not in the correct format.

File: src/server/utils.ts [hunk 11]
```diff
@@ -1,17 +1,15 @@
-import { readFile as fsReadFile, stat as fsStat } from 'fs';
-import { promisify } from 'util';
-
-export const readFile = promisify(fsReadFile);
-export const stat = promisify(fsStat);
-
 /**
  * Utility timer function for Nodejs performance profiling.
```
Comment (adamdbradley) [inline src/server/utils.ts:2]: Remove the `Nodejs` part of the comment

Comment (adamdbradley) [reply src/server/utils.ts:2]: Probably also state. "Returns a duration for `0` in environments that do not support `performance`."

File: src/testing/platform.ts [hunk 12]
```diff
@@ -1,9 +1,10 @@
-import { getPlatform, setPlatform } from '@builder.io/qwik';
+import { getPlatform, QRL, setPlatform } from '@builder.io/qwik';
 import type { TestPlatform } from './types';
 import { existsSync } from 'fs';
 import { fileURLToPath } from 'url';
+import { qExport } from '../core/import/qImport';
 
-export function setTestPlatform(document: any) {
+function createPlatform(document: Document) {
```
Comment (adamdbradley) [inline src/testing/platform.ts:7]: I left it as an `any` here cuz I'm not sure if like jsdom or domino's types would error against the lib.dom `Document`.

File: starters/apps/starter-builder/rollup.config.js [hunk 13]
```diff
@@ -1,25 +1,25 @@
...
+      'src/my-app.tsx'
     ],
     plugins: [
       nodeResolve(),
       qwikRollup({
-        symbolsPath: 'q-symbols.json',
-      }), 
-      typescript(),
+        symbolsOutput: (data) => {
+          outputJSON('./server/build/q-symbols.json', data);
```
Comment (adamdbradley) [inline starters/apps/starter-builder/rollup.config.js:14]: Where is `outputJSON` coming from?

File: starters/apps/starter-partytown/rollup.config.js [hunk 14]
```diff
@@ -1,25 +1,25 @@
...
+      'src/my-app.tsx'
     ],
     plugins: [
       nodeResolve(),
       qwikRollup({
-        symbolsPath: 'q-symbols.json',
-      }), 
-      typescript(),
+        symbolsOutput: (data) => {
+          outputJSON('./server/build/q-symbols.json', data);
```
Comment (adamdbradley) [inline starters/apps/starter-partytown/rollup.config.js:14]: Same, where's the import for `outputJSON`

File: starters/apps/starter/rollup.config.js [hunk 15]
```diff
@@ -1,25 +1,25 @@
...
+      'src/my-app.tsx'
     ],
     plugins: [
       nodeResolve(),
       qwikRollup({
-        symbolsPath: 'q-symbols.json',
-      }), 
-      typescript(),
+        symbolsOutput: (data) => {
+          outputJSON('./server/build/q-symbols.json', data);
```
Comment (adamdbradley) [inline starters/apps/starter/rollup.config.js:14]: `outputJSON`

File: starters/servers/cloudflare/rollup.config.server.js [hunk 16]
```diff
@@ -0,0 +1,30 @@
...
+      qwikRollup({
+        entryStrategy: {
+          type: 'single',
+        },
+      }),
+      commonjs(),
+    ],
+    output: [
+      {
+        intro: 'const global = globalThis;',
```
Comment (adamdbradley) [inline starters/servers/cloudflare/rollup.config.server.js:24]: `const global = globalThis;` can get removed from here now right?

File: starters/servers/cloudflare/wrangler.toml [hunk 17]
```diff
@@ -0,0 +1,18 @@
+name = "qwik-app"
```
Comment (adamdbradley) [inline starters/servers/cloudflare/wrangler.toml:1]: `qwik-project-name` right?

File: src/core/api.md (hunk 5 above)
Comment (manucorporat) [reply src/core/api.md:7]: I saw it too, I think its a bundling of types, going to dig deeper.

File: src/optimizer/core/src/transform.rs (hunk 6 above)
Comment (manucorporat) [reply src/optimizer/core/src/transform.rs:376]: yes! it was a stupid idea to include import.meta.url

File: src/optimizer/src/rollup/index.ts (hunk 7 above)
Comment (manucorporat) [reply src/optimizer/src/rollup/index.ts:143]: good one

File: starters/apps/starter-builder/rollup.config.js (hunk 13 above)
Comment (manucorporat) [reply starters/apps/starter-builder/rollup.config.js:14]: wow! nice catch

File: starters/servers/cloudflare/wrangler.toml (hunk 17 above)
Comment (manucorporat) [reply starters/servers/cloudflare/wrangler.toml:1]: yes!

File: starters/servers/cloudflare/rollup.config.server.js (hunk 16 above)
Comment (manucorporat) [reply starters/servers/cloudflare/rollup.config.server.js:24]: weird... I thought it did!

File: starters/servers/cloudflare/wrangler.toml (hunk 17 above)
Comment (manucorporat) [reply starters/servers/cloudflare/wrangler.toml:1]: need to modify the CLI to not copy... but read and write. cuz I don't have a way to find replace
in other files other than package.json

**Pull request merged**